src/ui/src/metadata_widget.cpp

```diff
@@ -1,10 +1,35 @@
 #include "metadata_widget.h"
 
-#include "ui_metadata_widget.h"
-
 MetadataWidget::MetadataWidget(QWidget* parent)
-		: QDockWidget(parent), ui_(new Ui::MetadataWidget) {
-	ui_->setupUi(this);
+		: QTableWidget(parent) {
+	
+	setColumnCount(2);
+	setHorizontalHeaderLabels(lables_);
+	// Test();
+}
+
+void MetadataWidget::OnShowMetadata(const MetadataPack& pack) {
+	setRowCount(pack.size());
+	size_t row_counter = 0;
+	for (const auto metadata : pack) {
+		QTableWidgetItem* meaning = new QTableWidgetItem(
+			QString::fromStdString(metadata.meaning()));
+		QTableWidgetItem* value =
+				new QTableWidgetItem(QString::fromStdString(metadata.value()));
+		setItem(row_counter, 0, meaning);
+		setItem(row_counter, 1, value);
```

Comment:
Narrowing conversion from size_t (unsigned long long) to int. Escape such conversions or make them explicit